enterprise/backend/src/metabase_enterprise/caching/api.clj

```diff
@@ -0,0 +1,118 @@
+(ns metabase-enterprise.caching.api
+  (:require
+   [compojure.core :refer [GET]]
+   [metabase.api.common :as api]
+   [metabase.api.common.validation :as validation]
+   [metabase.api.routes.common :refer [+auth]]
+   [metabase.events :as events]
+   [metabase.util :as u]
+   [metabase.util.cron :as u.cron]
+   [metabase.util.i18n :refer [tru]]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(defn- assert-valid-models [model ids]
+  (if (= model "root")
+    (when-not (zero? (first ids))
+      (throw (ex-info (tru "Root configuration is only valid with model_id = 0") {:status-code 400
+                                                                                  :model_id    (first ids)})))
+    (api/check-404 (t2/select-one (case model
+                                    "database"  :model/Database
+                                    "dashboard" :model/Dashboard
+                                    "question"  :model/Card)
+                                  :id [:in ids]))))
+
+(defn- audit-caching-change! [id target prev new]
+  (events/publish-event!
+   :event/caching-update
+   {:user-id api/*current-user-id*
+    :model   :model/CacheConfig
+    :details {:id             id
+              :model          (:model target)
+              :model_id       (:model_id target)
+              :previous-value prev
+              :next-value     new}}))
+
+(api/defendpoint GET "/"
+  "Return cache configuration."
+  [:as {{:strs [model collection]
+         :or   {model "root"}}
+        :query-params}]
+  {model      (ms/QueryVectorOf [:enum "root" "database" "dashboard" "question"])
+   collection [:maybe ms/PositiveInt]}
+  (validation/check-has-application-permission :setting)
+  (let [items (t2/select :model/CacheConfig
+                         :model [:in model]
+                         {:left-join [:report_card      [:and
+                                                         [:= :model [:inline "question"]]
+                                                         [:= :model_id :report_card.id]
+                                                         [:= :report_card.collection_id collection]]
+                                      :report_dashboard [:and
+                                                         [:= :model [:inline "dashboard"]]
+                                                         [:= :model_id :report_dashboard.id]
+                                                         [:= :report_dashboard.collection_id collection]]]
+                          :where     [:case
+                                      [:= :model [:inline "question"]]  [:!= :report_card.id nil]
+                                      [:= :model [:inline "dashboard"]] [:!= :report_dashboard.id nil]
+                                      :else                             true]})]
+    {:items (for [item items]
```

Comment:
It feels like a generic "key with list of stuffs" to me. I've just looked around - collection API uses `:data []` for that; maybe it makes sense to use the same key, I'll talk to frontend about that.

Reply:
Happy with something generic if it's standarized on the frontend. Otherwise I'd prefer we find a consistent name we can use for variables holding these kinds of values, without any ambiguity over meaning.

Reply:
Renamed to `:data` after discussion with Raffi.